pandas/core/internals.py

```diff
@@ -2914,37 +2914,54 @@ def sparse_reindex(self, new_index):
                                           placement=self.mgr_locs)
 
 
+_block_type_map = {
```

Comment:
rather than constructing this manually and adding code, we *already* have a mapping by using the block class name, or could add shortnames to each class. In any event, this can be autmatically constructed by interation over the blocks (you have to do this at the end of the file). 

Alternatively, can have a registry (a dict), that each class registers itself), but that is slightly more complicated.

Reply:
> Alternatively, can have a registry (a dict), that each class registers itself), but that is slightly more complicated.

That's what I have in mind longer-term, which is why I went with a module-level dict for this.  But you're right that is ways away.

> rather than constructing this manually and adding code, we already have a mapping by using the block class name, or could add shortnames to each class. In any event, this can be autmatically constructed by interation over the blocks (you have to do this at the end of the file).

Are you thinking something like:

```
globs = globals()
_block_type_map = {x: globs[x] for x in globs if inspect.issclass(globs[x]) and issubclass(globs[x], Block)}
```

If so, this is one of the class of things that I'll begrudgingly implement if you insist.

Reply:
not at all, simply define

``register_block`` as a function which sets the ``registry = {}`` to ``Block.name -> Block``
then this is pretty easy